test.js

```diff
@@ -45,3 +47,29 @@ test('is mobile', function (t) {
 
   t.end()
 })
+
+describe('ua-bruteforce', function () {
+  const limit = 300
+  const checks = {
+    mobile: { result: true },
+    tablet: { result: true, tablet: true },
+    desktop: { result: false }
+  }
+  const testCases = Object.keys(checks).reduce((cases, deviceCategory) => {
+    cases.push(
+      ...new Array(limit).fill().map(() => ({
+        ua: new UserAgent({ deviceCategory }).toString(),
+        ...checks[deviceCategory]
+      }))
+    )
+
+    return cases
+  }, [])
```

Comment:
```suggestion
  const testCases = Object
    .entries(checks)
    .reduce((cases, [deviceCategory, { result, tablet }]) => ([
      ...cases,
      ...new Array(limit).fill().map(() => ({
        ua: new UserAgent({ deviceCategory }).toString(),
        result,
        tablet
      }))
    ]), [])
```

- Cleaner lookup using `Object.entries`
- use `...` to simplify the logic
- make explicit which properties get added

Reply:
Agree. This is really much better.